cdk/kittyhawk/src/ingress.ts

```diff
@@ -13,55 +13,101 @@ export interface IngressProps {
-        if (apex_domain) {
-          const host_string = apex_domain[0].split('.').join('-').concat('-tls');
-          return { hosts: [h.host], secretName: host_string };
-        } else {throw new Error(`Ingress construction failed: apex domain regex failed on ${h}`);}
-      });
+    const tls = props.rules.map(h => {
+      const hostString: string = `${domainToCertName(h.host, h.isSubdomain ?? false)}-tls`;
+      return { hosts: [h.host], secretName: hostString };
+    });
 
-      let rules = ingress.map(h => {
-        return {
-          host: h.host,
-          http: {
-            paths: h.paths.map(path => {
-              return {
-                path: path,
-                backend: {
-                  serviceName: appname,
-                  servicePort: IntOrString.fromNumber(port),
+    const rules: IngressRule[] = props.rules.map(h => {
+      return {
+        host: h.host,
+        http: {
+          paths: h.paths.map(path => ({
+            path: path,
+            pathType: 'Prefix',
+            backend: {
+              service: {
+                name: appname,
+                port: {
+                  number: fullConfig.port,
                 },
-              };
-            }),
-          },
-        };
-      });
-
-      new IngressApiObject(this, `ingress-${appname}`, {
-        metadata: {
-          name: appname,
-          namespace: 'default',
-        },
-        spec: {
-          tls,
-          rules,
+              },
+            },
+          })),
         },
-      });
+      };
+    });
+
+    new IngressApiObject(this, `ingress-${appname}`, {
+      metadata: {
+        name: appname,
+        namespace: 'default',
+      },
+      spec: {
+        tls,
+        rules,
+      },
+    });
+  }
+}
+
+/**
+ * Removes the subdomain from an url if isSubdomain is true
+ * @param d the domain as a string
+ * @param isSubdomain true if the url is a subdomain of a domain that already has a certificate; default to false if unspecified.
+ */
+export function removeSubdomain(d: string, isSubdomain = false) {
+  if (isSubdomain) {
+    // Must have at least 3 parts to the domain (e.g. xxx.abc.com)
+    if (d.split('.').length < 3) {
+      throw new Error(`No subdomain found in ${d}.`);
     }
+    return d.split('.').slice(1).join('.');
+  } else {
+    return d;
   }
-}
\ No newline at end of file
+}
+
+/**
+ * Converts a domain to a dash-separated form (e.g. abc-def-org), optionally removing the subdomain.
+ * @param d the domain as a string
+ * @param isSubdomain true if the url is a subdomain of a domain that already has a certificate; default to false if unspecified.
+ */
+export function domainToCertName(d: string, isSubdomain = false) {
+  // Remove everything before the 1st '.' if it is a subdomain.
+  if (d.split('.').length < 2) {
+    throw new Error(`Ingress creation failed: domain ${d} is invalid.`);
+  }
+  return removeSubdomain(d, isSubdomain).split('.').join('-');
```

Comment:
I think this and `domainToCertName` should accept inputs `d` and `isSubdomain` in the form of:

```
export function domainToCertName({d, isSubdomain = false}: {d: string, isSubdomain: boolean}) {
```

Same thing applies to `removeSubdomain`. Right now, when we call the function, we call `domainToCertName("blah", true)`, which doesn't make a lot of sense because we have no idea what "true" means unless we look at the function itself. 

If we do it in the way outlined above, we would call `domainToCertName({d: "blah", isSubdomain: true})` which is much cleaner. Thoughts?

Reply:
Talked to Ying & Armaan and decided to not do this